Unexpose event.relatedTarget when shadow tree state has changed

nt1m opened the pull request:
#### 73d8160d105b0b8e5e62d397e6e4f67956b6a984
<pre>
Unexpose event.relatedTarget when shadow tree state has changed
<a href="https://bugs.webkit.org/show_bug.cgi?id=242671">https://bugs.webkit.org/show_bug.cgi?id=242671</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/dom/EventContext.cpp:
(WebCore::EventContext::handleLocalEvents const):
* Source/WebCore/dom/EventContext.h:
(WebCore::EventContext::setRelatedTarget):
* Source/WebCore/dom/EventPath.cpp:
(WebCore::RelatedNodeRetargeter::RelatedNodeRetargeter):
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::moveTreeToNewScope):
* Source/WebCore/dom/Node.h:
(WebCore::Node::hasBeenInUserAgentShadowTree const):
* Source/WebCore/dom/ShadowRoot.cpp:
(WebCore::ShadowRoot::ShadowRoot):

Canonical link: <a href="https://commits.webkit.org/252631@main">https://commits.webkit.org/252631@main</a>
</pre>

rniwa reviewed (changes requested):
As we discussed elsewhere, I don't think this change is right.

rniwa reviewed (approved):
r=me assuming no perf regression.

nt1m committed:
Unexpose event.relatedTarget when shadow tree state has changed
https://bugs.webkit.org/show_bug.cgi?id=242671

Reviewed by Ryosuke Niwa.

* Source/WebCore/dom/EventContext.cpp:
(WebCore::EventContext::handleLocalEvents const):
* Source/WebCore/dom/EventContext.h:
(WebCore::EventContext::setRelatedTarget):
* Source/WebCore/dom/EventPath.cpp:
(WebCore::RelatedNodeRetargeter::RelatedNodeRetargeter):
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::moveTreeToNewScope):
* Source/WebCore/dom/Node.h:
(WebCore::Node::hasBeenInUserAgentShadowTree const):
* Source/WebCore/dom/ShadowRoot.cpp:
(WebCore::ShadowRoot::ShadowRoot):

Canonical link: https://commits.webkit.org/252631@main
